hpy/debug/src/debug_ctx_cpython.c

```diff
@@ -78,8 +79,13 @@ void debug_ctx_CallRealFunctionFromTrampoline(HPyContext *dctx,
         HPyFunc_noargs f = (HPyFunc_noargs)func;
         _HPyFunc_args_NOARGS *a = (_HPyFunc_args_NOARGS*)args;
         DHPy dh_self = _py2dh(dctx, a->self);
-        DHPy dh_result = f(dctx, dh_self);
-        DHPy_close_and_check(dctx, dh_self);
+        ((HPyDebugInfo*)dctx->_private)->is_active = false;
+        HPyContext *dctx_copy = malloc(sizeof(struct _HPyContext_s));
```

Comment:
Probably worth checking if malloc succeeded and raising HPyErr_NoMemory if not. See:
https://github.com/hpyproject/hpy/blob/7cae5beda73f28dbeb0b826e0082a92fda4dd547/hpy/devel/src/runtime/ctx_tracker.c#L88-L92